solr/solrj/src/java/org/apache/solr/client/solrj/impl/HttpClientBuilderFactory.java

```diff
@@ -19,10 +19,7 @@
 import java.io.Closeable;
 
 /**
- * Factory interface for configuring {@linkplain SolrHttpClientBuilder}. This relies on the internal
- * HttpClient implementation and is subject to change.
- *
- * @lucene.experimental
+ * Factory interface for configuring {@linkplain SolrHttpClientBuilder}.
```

Comment:
@dsmiley What is your take on `HttpClientBuilderFactory`? It is used to build the old Apache Client but has a `setup` method relying on `Http2SolrClient`. So do we classify this class as a `apache-client` or `jetty-client` class, or both?

Perhaps we keep the experimental tag on this class, since its API may need to change to separate the two?

Reply:
Firstly; IMO `HttpClientBuilderFactory` and `SolrHttpClientBuilder` ought not to be in SolrJ because they are only used for inter-node communication within Solr.  It's a distraction for our users to see stuff like that in SolrJ.

Indeed `HttpClientBuilderFactory` will change so keeping experimental there is right.  An aside: I'd rather see us do the inverse of "lucene.experimental" -- have "solr.stable".  If "solr.stable" is absent from a class; watch out!

Reply:
> Firstly; IMO `HttpClientBuilderFactory` and `SolrHttpClientBuilder` ought not to be in SolrJ because they are only used for inter-node communication within Solr.

Good point, I'll look into this separately

Reply:
Ah.  Well before long (by v10), we'll remove the Apache HttpClient support in these specific abstractions on main since that client will cease to exist.